### ISSUE TITLE
Why can't I always predict the left boundary when performing instance segmentation?

### ISSUE THREAD

**Issue by zwf159** [ultralytics/ultralytics#15610]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

为什么我在预测的时候把边界框关闭了之后我左边的边界总是会留一个位置？zh只有左边有这个问题
![image_2485](https://github.com/user-attachments/assets/4da9c2e6-12c0-45f4-bf56-403e84a17b37)

**Comment by Y-T-G** [comment]
You can add a margin.

https://github.com/ultralytics/ultralytics/issues/14365#issuecomment-2225105580

**Comment by zwf159** [comment]
> 您可以添加边距。
> 
> [#14365 （评论）](https://github.com/ultralytics/ultralytics/issues/14365#issuecomment-2225105580)

![image_2482](https://github.com/user-attachments/assets/2fb7fdb7-3cc4-46e2-b3fa-1075815a67aa)
His response did not solve my problem; the previous image shows the result after using his code.

**Comment by glenn-jocher** [comment]
@zwf159 thank you for the update. To better assist you, could you please provide more details about your current setup and the specific code you are using? This will help us diagnose the issue more accurately.

**Comment by zwf159** [comment]
> @zwf159 thank you for the update. To better assist you, could you please provide more details about your current setup and the specific code you are using? This will help us diagnose the issue more accurately.
    Thank you for reply.It's like this: as shown in the first question, there is a spot on the left side of my mask that hasn't been covered. This seems to be a bug because I have labeled all my images in this manner, and there shouldn't be any issues. After using a solution referenced by [Y-T-G](https://github.com/Y-T-G), it resolved part of the problem, but the middle part on the left side of my second image still isn't covered. I have adjusted the ratio, but I still get the same response. This is the code I used:
from ultralytics import YOLO
import glob
import os
import torch
import ultralytics.utils.ops as ops

def crop_mask(masks, boxes, margin=0.05):
    n, h, w = masks.shape
    x1, y1, x2, y2 = boxes.T

    # Expand only the left boundary (x1)
    x1 = (x1 - margin * (x2 - x1)).clamp(0, w)

    # Keep other boundaries unchanged
    boxes = torch.stack((x1, y1, x2, y2), dim=1)

    # Adjust mask position
    masks = masks[:, y1.int():y2.int(), x1.int():x2.int()]
    return masks

ops.crop_mask = crop_mask

model = YOLO(r'./runs/segment/train17/weights/best.pt')  # load an official model

# Predict with the model
imgpath = r'./test21'
imgs = glob.glob(os.path.join(imgpath, '*.png'))
for img in imgs:
    model.predict(img, save=True, conf=0.6, retina_masks=False, augment=False, show_labels=False, imgsz=640, device='0', max_det=1, show_boxes=False)

**Comment by zwf159** [comment]
> @zwf159 thank you for the update. To better assist you, could you please provide more details about your current setup and the specific code you are using? This will help us diagnose the issue more accurately.



    Thank you for reply.It's like this: as shown in the first question, there is a spot on the left side of my mask that hasn't been covered. This seems to be a bug because I have labeled all my images in this manner, and there shouldn't be any issues. After using a solution referenced by [Y-T-G](https://github.com/Y-T-G), it resolved part of the problem, but the middle part on the left side of my second image still isn't covered. I have adjusted the ratio, but I still get the same response. This is the code I used:
from ultralytics import YOLO
import glob
import os
import torch
import ultralytics.utils.ops as ops

def crop_mask(masks, boxes, margin=0.05):
    n, h, w = masks.shape
    x1, y1, x2, y2 = boxes.T

    # Expand only the left boundary (x1)
    x1 = (x1 - margin * (x2 - x1)).clamp(0, w)

    # Keep other boundaries unchanged
    boxes = torch.stack((x1, y1, x2, y2), dim=1)

    # Adjust mask position
    masks = masks[:, y1.int():y2.int(), x1.int():x2.int()]
    return masks

ops.crop_mask = crop_mask

model = YOLO(r'./runs/segment/train17/weights/best.pt')  # load an official model

# Predict with the model
imgpath = r'./test21'
imgs = glob.glob(os.path.join(imgpath, '*.png'))
for img in imgs:
    model.predict(img, save=True, conf=0.6, retina_masks=False, augment=False, show_labels=False, imgsz=640, device='0', max_det=1, show_boxes=False)

**Comment by zwf159** [comment]
> Thank you for reply.It's like this: as shown in the first question, there is a spot on the left side of my mask that hasn't been covered. This seems to be a bug because I have labeled all my images in this manner, and there shouldn't be any issues. After using a solution referenced by [Y-T-G](https://github.com/Y-T-G), it resolved part of the problem, but the middle part on the left side of my second image still isn't covered. I have adjusted the ratio, but I still get the same response. This is the code I used:

Thank you for reply.It's like this: as shown in the first question, there is a spot on the left side of my mask that hasn't been covered. This seems to be a bug because I have labeled all my images in this manner, and there shouldn't be any issues. After using a solution referenced by [Y-T-G](https://github.com/Y-T-G), it resolved part of the problem, but the middle part on the left side of my second image still isn't covered. I have adjusted the ratio, but I still get the same response. This is the code I used:

**Comment by glenn-jocher** [comment]
Thank you for the detailed explanation. Please ensure you are using the latest version of the Ultralytics package. If the issue persists, consider opening a new issue on GitHub with your complete code and sample images for further investigation.
